### PR TITLE
SiteSync: fix transitive alternate sites, fix dropdown in Local Settings

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Based on specific use case for a customer where separate site is used for specific project.

## Description
'studio2' site was created to separate synchronization for specific project, previous implementation didn't handle setting this site as an alternative site to `studio`.

## Testing notes:
1. this PR will be tested by customer on their specific configuration
2. It should be tested on regular configuration if it didn't break anything (just pulled and start OP and try to publish something if it behaves same way as before)

### CODE REVIEW

File: openpype/plugins/publish/integrate_new.py [hunk 1]
```diff
@@ -1158,22 +1157,65 @@ def _add_alternative_sites(self,
...
+            for alt_site in alt_sites:
+                if not alt_site_pairs.get(alt_site):
+                    alt_site_pairs[alt_site] = []
+                alt_site_pairs[alt_site].extend([site_name])
+
+        # transitive relationship, eg site is alternative to another which is
+        # alternative to nex site
+        loop = True
+        while loop:
+            loop = False
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_new.py:1204]: What is this `loop` for? :D

Comment (BigRoy) [reply openpype/plugins/publish/integrate_new.py:1204]: Or potentially easier: https://stackoverflow.com/a/41553897/1838864

Comment (iLLiCiTiT) [reply openpype/plugins/publish/integrate_new.py:1204]: Or
```
for site_name, alt_sites in alt_site_pairs.items():
    sites_queue = collections.deque()
    sites_queue.extend(alt_sites)
    while sites_queue:
        alt_site = sites_queue.popleft()

        # safety against wrong config
        # {"SFTP": {"alternative_site": "SFTP"}
        if alt_site == site_name or alt_site not in alt_site_pairs:
            continue

        for alt_alt_site in alt_site_pairs[alt_site]:
            if (
                alt_alt_site != site_name
                and alt_alt_site not in alt_sites
            ):
                alt_sites.append(alt_alt_site)
                sites_queue.append(alt_alt_site)
```

**Comment by kalisp** [comment]
It is working for customer.

File: openpype/plugins/publish/integrate_new.py [hunk 2]
```diff
@@ -1158,22 +1157,65 @@ def _add_alternative_sites(self,
...
-                    real_created = already_attached_sites[added_site]
-                    # alt site inherits state of 'created_dt'
-                    if real_created:
-                        meta["created_dt"] = real_created
-                    rec["sites"].append(meta)
-                    already_attached_sites[meta["name"]] = real_created
 
-        return rec
+                    for alt_alt_site in alt_site_pairs.get(alt_site, []):
+                        if (    alt_alt_site != site_name
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply openpype/plugins/publish/integrate_new.py:1213]: I agree with hound

File: openpype/plugins/publish/integrate_new.py [hunk 3]
```diff
@@ -1158,22 +1158,67 @@ def _add_alternative_sites(self,
...
+        """Returns dict of site and its alternative sites.
+
+        If `site` has alternative site, it means that alt_site has 'site' as
+        alternative site
+        Args:
+            conf_sites (dict)
+        Returns:
+            (dict): {'site': [alternative sites]...}
+        """
+        alt_site_pairs = {}
```
Comment (iLLiCiTiT) [inline openpype/plugins/publish/integrate_new.py:1188]: I would change this to `collections.defaultdict(list)`, you could remove all of the `if not alt_site_pairs.get(...` checks.

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**